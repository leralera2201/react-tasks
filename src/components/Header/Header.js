import React from "react";
import { Navbar, NavbarBrand } from "reactstrap";
import { Link, NavLink } from "react-router-dom";

import './Header.scss';

const Header = () => {
  return (
    <Navbar color="dark" dark expand="md" className="h50">
      <div className="container flex-space-between">
        <Link to={"/"}>
          <NavbarBrand className="fz30 p0">Blog</NavbarBrand>
        </Link>
        <NavLink
          exact
          activeClassName="link-active"
          to="/books"
          className="fz20 link"
        >
          Book
        </NavLink>
      </div>
    </Navbar>
  );
};

export default Header;
