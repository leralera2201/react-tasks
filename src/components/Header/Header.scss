.link {
  color: white;
  padding: 0 10px;
  &-active {
    background: white;
    color: black !important;
  }
  &:hover {
    color: white;
  }
}
