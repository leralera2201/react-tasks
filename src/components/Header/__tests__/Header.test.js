import React from 'react';
import { BrowserRouter as Router} from 'react-router-dom'
import { render } from "enzyme";
import toJson from "enzyme-to-json";

import Header from "../Header";

describe("Testing Header component", () => {

    it('renders correctly', () => {
        const wrapper = render(<Router><Header /></Router>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

})