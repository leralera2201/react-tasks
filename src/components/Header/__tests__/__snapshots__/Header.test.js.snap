// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header component renders correctly 1`] = `
<nav
  class="h50 navbar navbar-expand-md navbar-dark bg-dark"
>
  <div
    class="container flex-space-between"
  >
    <a
      class="fz30 p0 header-logo color-f"
      href="/"
    >
      Blog
    </a>
    <a
      class="fz20 link"
      href="/books"
    >
      Book
    </a>
  </div>
</nav>
`;
